src/H5FDmulti.c

```diff
@@ -35,6 +35,25 @@
 #define TRUE 1
 #endif
 
+/* Imported from H5private.h */
+/* Macros for enabling/disabling particular GCC warnings */
+/* (see the following web-sites for more info:
+ *      http://www.dbp-consulting.com/tutorials/SuppressingGCCWarnings.html
+ *      http://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html#Diagnostic-Pragmas
+ */
+/* These pragmas are only implemented usefully in gcc 4.6+ */
+#if ((__GNUC__ * 100) + __GNUC_MINOR__) >= 406
+#define H5_GCC_DIAG_JOINSTR(x, y) x y
+#define H5_GCC_DIAG_DO_PRAGMA(x)  _Pragma(#x)
+#define H5_GCC_DIAG_PRAGMA(x)     H5_GCC_DIAG_DO_PRAGMA(GCC diagnostic x)
+
+#define H5_GCC_DIAG_OFF(x) H5_GCC_DIAG_PRAGMA(push) H5_GCC_DIAG_PRAGMA(ignored H5_GCC_DIAG_JOINSTR("-W", x))
+#define H5_GCC_DIAG_ON(x)  H5_GCC_DIAG_PRAGMA(pop)
+#else
+#define H5_GCC_DIAG_OFF(x)
+#define H5_GCC_DIAG_ON(x)
+#endif
+
```

Comment:
This does not belong in the public header. We can move it to a separate header file if we really care about the code duplication.

Reply:
I agree with you - this shouldn't be in H5public.h.   I doubt H5warning.h would work though, since this VFD requires a public header and I don't think H5warning.h is great for a public header.   Probably duplicating the code is fine, although I would add a comment here and in H5private.h that the macros should be kept the same.

Also - I wouldn't use the plain "H5" prefix here, but probably "H5FD_MULTI" or something similar.

Reply:
Good point. I'll change that the next time I make changes to this PR.